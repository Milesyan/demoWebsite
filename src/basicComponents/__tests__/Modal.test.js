/* eslint-disable no-undef */ /*eslint-disable react/react-in-jsx-scope*/

import { Modal } from '../index';

const setup = overrideProps => {
  const initProps = {
    onClick: jest.fn()
  };
  const props = { ...initProps, ...overrideProps};
  const wrapper = shallow(
    <Modal {...props}>
      <div>
        TEST CHILDREN CONTENT
      </div>
    </Modal>
  );
  return { wrapper, props };
}

describe('Modal snapshots', () => {

  it('should match snapshot when Modal is not disabled', () => {
    const { wrapper } = setup();
    expect(wrapper).toMatchSnapshot();
  });

});

describe('Modal functions', () => {

  it('should call the onClick function on the outside mask', () => {
    const { wrapper, props } = setup();
    wrapper.simulate('click')
    expect(props.onClick).toBeCalled();
  });
  
});