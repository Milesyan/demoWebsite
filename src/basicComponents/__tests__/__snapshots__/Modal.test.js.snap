// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal snapshots should match snapshot when Modal is not disabled 1`] = `
<CSSTransitionGroup
  transitionAppear={true}
  transitionAppearTimeout={500}
  transitionEnter={false}
  transitionLeave={false}
  transitionName={
    Object {
      "appear": "appear",
      "appearActive": "appearActive",
    }
  }
>
  <div
    className="container"
    onClick={[MockFunction]}
  >
    <div
      className="main"
      onClick={[Function]}
    >
      <div>
        TEST CHILDREN CONTENT
      </div>
    </div>
  </div>
</CSSTransitionGroup>
`;
